salt/states/user.py

```diff
@@ -490,22 +567,41 @@ def present(name,
             groups.extend(present_optgroups)
         elif present_optgroups:
             groups = present_optgroups[:]
-        if __salt__['user.add'](name,
-                                uid=uid,
-                                gid=gid,
-                                groups=groups,
-                                home=home,
-                                shell=shell,
-                                unique=unique,
-                                system=system,
-                                fullname=fullname,
-                                roomnumber=roomnumber,
-                                workphone=workphone,
-                                homephone=homephone,
-                                loginclass=loginclass,
-                                createhome=createhome):
+
+        # Setup params specific to Linux and Windows to be passed to the
+        # add.user function
+        if not salt.utils.is_windows():
```

Comment:
Can you extract this section into a helper function? This will help unit testing a lot.